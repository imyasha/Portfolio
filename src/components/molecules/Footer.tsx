import React from "react"
import styled from "styled-components"

const Foot = styled.footer`
  padding: 0 3%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

const Footer = () => {
  return (
    <Foot>
      <div>
        &copy; 2019 Jacob Herper
        <br />
        Proudly built with{" "}
        <a href="https://gatsbyjs.org/" target="_blank">
          Gatsby
        </a>
        .
      </div>
      <div>Social Links ...</div>
    </Foot>
  )
}

export default Footer
