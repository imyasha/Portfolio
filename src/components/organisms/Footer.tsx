import React from 'react';
import styled from 'styled-components';
import Menu from 'Atoms/Menu';

const Foot = styled.footer`
  padding: 0 3% 1rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  @media print {
    display: none;
  }

  @media screen and (max-width: 768px) {
    flex-direction: column;
    text-align: center;
  }
`;

const Footer = () => (
  <Foot>
    <div>
      &copy; {new Date().getFullYear()} Jacob Herper
      <br />
      Proudly built with{' '}
      <a href="https://gatsbyjs.org/" target="_blank" rel="noopener noreferrer">
        Gatsby
      </a>
      .
    </div>
    <nav>
      <Menu data-testid="footer-menu">
        <li data-testid="github">
          <a
            href="https://github.com/jakeherp"
            target="_blank"
            rel="noopener noreferrer"
          >
            Github
          </a>
        </li>
        <li data-testid="linkedin">
          <a
            href="https://www.linkedin.com/in/jacobherper/"
            target="_blank"
            rel="noopener noreferrer"
          >
            LinkedIn
          </a>
        </li>
        <li data-testid="twitter">
          <a
            href="https://twitter.com/jakeherp"
            target="_blank"
            rel="noopener noreferrer"
          >
            Twitter
          </a>
        </li>
      </Menu>
    </nav>
  </Foot>
);

export default Footer;
