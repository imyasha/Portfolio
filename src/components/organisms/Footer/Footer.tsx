import { SocialIcons } from 'Molecules/SocialIcons';

const Footer = () => {
	return (
		<footer className="flex justify-between mt-8 md:mt-20 px-4 md:px-20 py-8">
			<span>&copy; {new Date().getFullYear()} Jacob Herper</span>
			<SocialIcons
				profiles={[
					{
						name: 'Github',
						url: 'https://www.github.com/jakeherp',
						icon: 'GITHUB',
					},
					{
						name: 'LinkedIn',
						url: 'https://www.linkedin.com/in/jacobherper',
						icon: 'LINKEDIN',
					},
					{
						name: 'Twitter',
						url: 'https://twitter.com/jakeherp',
						icon: 'TWITTER',
					},
					{
						name: 'Instagram',
						url: 'https://www.instagram.com/jakeherp',
						icon: 'INSTAGRAM',
					},
				]}
			/>
		</footer>
	);
};

export { Footer };
