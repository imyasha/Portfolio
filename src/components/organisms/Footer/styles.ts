import styled from 'styled-components';

const StyledFooter = styled.footer`
	padding: 2rem 0;
`;

const Flex = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column-reverse;

	@media screen and (min-width: 768px) {
		flex-direction: row;
		justify-content: space-between;
	}
`;

export { Flex, StyledFooter };
