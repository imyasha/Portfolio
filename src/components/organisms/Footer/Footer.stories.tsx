import { Footer } from './Footer';

export default {
	title: 'organisms/Footer',
	component: Footer,
};

export const FooterStory = () => <Footer />;

FooterStory.storyName = 'Footer';
