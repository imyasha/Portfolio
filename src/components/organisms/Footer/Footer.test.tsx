import { Footer } from '../Footer';
import { render } from '@testing-library/react';

describe('Footer', () => {
	it('renders correctly', () => {
		const { container } = render(<Footer />);
		expect(container).toMatchSnapshot();
	});
});
