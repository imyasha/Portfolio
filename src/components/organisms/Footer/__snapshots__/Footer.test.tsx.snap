// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <footer
    class="styles__StyledFooter-razpbj-0 jZveNS"
  >
    <div
      class="styles__StyledContainer-mepagl-0 exxZos"
    >
      <div
        class="styles__Flex-razpbj-1 gJdgaN"
      >
        <span>
          Â© 
          2021
           Jacob Herper
        </span>
        <ul
          class="styles__StyledSocialIcons-sc-15vq785-0 iHrrTj"
        >
          <li>
            <a
              href="https://github.com/jakeherp"
              rel="noopener noreferrer"
              target="_blank"
              title="Github"
            >
              <test-file-stub />
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/jakeherp"
              rel="noopener noreferrer"
              target="_blank"
              title="Twitter"
            >
              <test-file-stub />
            </a>
          </li>
          <li>
            <a
              href="https://www.linkedin.com/in/jacobherper"
              rel="noopener noreferrer"
              target="_blank"
              title="LinkedIn"
            >
              <test-file-stub />
            </a>
          </li>
          <li>
            <a
              href="https://instagram.com/jakeherp"
              rel="noopener noreferrer"
              target="_blank"
              title="Instagram"
            >
              <test-file-stub />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</div>
`;
